tree/treeplayer/src/TDFNodes.cxx

```diff
@@ -442,6 +442,65 @@ void TLoopManager::Report() const
    for (const auto &fPtr : fBookedNamedFilters) fPtr->PrintReport();
 }
 
+void TLoopManager::AddColumnAlias(std::string_view alias, std::string_view colName)
+{
+   auto aliasAsString = std::string(alias);
+
+   // Helper to find out if a name is a column
+   auto dsColumnNames = fDataSource ? &fDataSource->GetColumnNames() : nullptr;
+   auto branches = fTree ? fTree->GetListOfBranches() : nullptr;
+   auto nameIsCol = [&] (const std::string &name) {
+      auto isCustomColumn = std::find(fCustomColumnNames.begin(), fCustomColumnNames.end(), name) != fCustomColumnNames.end();
+      auto isDSColumn = dsColumnNames ? std::find(dsColumnNames->begin(), dsColumnNames->end(), name) != dsColumnNames->end() : false;
+      auto isBranch = branches ? (nullptr == branches->FindObject(name.c_str())) : false;
+      return isCustomColumn || isDSColumn || isBranch;
+   };
+
+   // If the alias name is a column name, there is a problem
+   if (nameIsCol(aliasAsString)) {
+      std::string msg = "The column alias ";
+      msg += alias;
+      msg += " is already the name of a column.";
+      throw std::runtime_error(msg);
+   }
+
+   auto colNameAsString = std::string(colName);
+   const auto aliasColumnNameMapEnd = fAliasColumnNameMap.end();
+   auto aliasColumnNameIt = fAliasColumnNameMap.find(colNameAsString);
+   auto isAliasToAlias = aliasColumnNameMapEnd != aliasColumnNameIt;
+
+   // If the alias is not in defined and the column name does not exist, there is a problem
+   if (!isAliasToAlias && !nameIsCol(colNameAsString)) {
+      std::string msg = "The column name ";
+      msg += colName;
+      msg += " is neither an existing column nor an alias.";
+      throw std::runtime_error(msg);
+   }
+
```

Comment:
I _think_ we can use `CheckCustomColumn` for all the error checking up to here -- it also deals with data-source columns.

Reply:
Yes. With some refactoring we can piggy back on the old routine which is absolutely a must - TDF is growing big enough w/o code duplication :sweat_smile:

Reply:
By that I meant, "done".